src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Modal, Col, Button, Alert } from 'patternfly-react';
+
+import { FormFactory } from '../Form/FormFactory';
+import { createBaremetalHost } from '../../k8s/request';
+import { TinyInlineLoading } from '../Loading/Loading';
+
+const formFields = {
+  name: {
+    id: 'name',
+    title: 'Name',
+    required: false,
+  },
+  controller: {
+    id: 'controller',
+    title: 'Management Controller Address',
+    required: true,
+  },
+  username: {
+    id: 'username',
+    title: 'Username',
+    required: true,
+  },
+  password: {
+    id: 'password',
+    title: 'Password',
+    type: 'password',
+    required: true,
+  },
+};
+
+const formText = `Specify the host's name, management controller IP address,
+and credentials to add it to the cluster.`;
+
+export class CreateBaremetalHostDialog extends React.Component {
+  state = {
+    form: {
+      value: {},
+      valid: false,
+    },
+    isSubmitting: false,
+    errorMessage: null,
+  };
+
+  onFormChange = (newValue, target, formValid) => {
+    this.setState(state => {
+      const form = { ...state.form };
+      form.value[target] = {
+        ...newValue,
+      };
+      form.valid = formValid;
+      return { form, errorMessage: null };
+    });
+  };
+
+  onSubmit = async () => {
+    this.setState(state => ({
+      ...state,
+      isSubmitting: true,
+    }));
+
+    return createBaremetalHost(this.props.k8sCreate, this.state.form.value, this.props.selectedNamespace)
+      .then(result => {
+        this.setState(state => ({
+          ...state,
+          isSubmitting: false,
+        }));
+        this.props.onClose();
+        return result;
+      })
+      .catch(({ message }) => {
+        this.setState(state => ({
+          ...state,
+          isSubmitting: false,
+          errorMessage: message,
+        }));
+      });
+  };
+
+  render() {
+    return (
+      <Modal show bsSize="large" onHide={this.props.onClose} className="kubevirt-create-baremetal-host-dialog">
+        <Modal.Header closeButton>
+          <Modal.Title>Add Host</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Col sm={12}>
```

Comment:
what is the reason for `Col` ?

Reply:
It makes the paragraph text at the top of the form to be aligned with the form fields; it receives some margin on the left.